### PR TITLE
add some spatial column support for create table with vtexplain

### PR DESCRIPTION
**Pull request by slanning**
This adds "let vtexplain parse it" support for spatial column types:
* GEOMETRY
* POINT
* LINESTRING
* POLYGON
* GEOMETRYCOLLECTION
* MULTIPOINT
* MULTILINESTRING
* MULTIPOLYGON

as well as SPATIAL INDEX in CREATE TABLE.

The spatial data types aren't fully documented:
https://dev.mysql.com/doc/refman/8.0/en/spatial-type-overview.html
In particular, there seems to be no grammar defined for it
(starting at https://dev.mysql.com/doc/refman/8.0/en/create-table.html
 its "spatial_type" is never specified as far as I can tell).
There are a couple of examples with POINT and GEOMETRY,
so I guessed the rest.
(GEOMETRY seems to be a generic type that can hold POINT, LINESTRING, and POLYGON;
 while GEOMETRYCOLLECTION holds multiple of any of them,
 and the MULTI* types restrict to only multiple of POINT, LINESTRING, or POLYGON.)
MySQL 8.0 adds the ability to specify RSID in the column definition,
but I didn't add anything for that (it's not in 5.7).

I assumed sqltypes.Geometry for SQLType based on
https://dev.mysql.com/doc/refman/5.7/en/gis-data-formats.html#gis-internal-format

SPATIAL INDEX is similar to "UNIQUE INDEX".
(I added "Spatial" to the IndexInfo struct but didn't use it anywhere;
 not sure if that's correct to do.)

Signed-off-by: Scott Lanning <scott.lanning@booking.com>

### CODE REVIEW

File: go/vt/sqlparser/ast.go [hunk 1]
```diff
@@ -1045,6 +1048,22 @@ func (ct *ColumnType) SQLType() querypb.Type {
...
+	case keywordStrings[POLYGON]:
+		return sqltypes.Geometry
+	case keywordStrings[GEOMETRYCOLLECTION]:
+		return sqltypes.Geometry
+	case keywordStrings[MULTIPOINT]:
+		return sqltypes.Geometry
+	case keywordStrings[MULTILINESTRING]:
+		return sqltypes.Geometry
+	case keywordStrings[MULTIPOLYGON]:
+		return sqltypes.Geometry
```
Comment (sougou) [inline go/vt/sqlparser/ast.go]: @demmer: I think this big switch can be avoided by:
1. `switch keywords[ct.Type]`
2. Building a `map[int]querypb.Type`

Not a blocker for this.

**Comment by sougou** [comment]
@slanning can you resolve the conflicts?

**Comment by slanning** [comment]
Not sure if that was correctly resolved. I think I botched merging master at first (not sure I even know how to do that...), then ended up just `git reset --hard` to where upstream master is and re-committed on top of that, so kind of manually rebased..

**Comment by sougou** [comment]
Missing DCO :)

**Comment by slanning** [comment]
It has the "Signed-off-by:" there, so I don't know why it's not working.
I'm out today, so I'll probably just end up making a new pull request next week.

**Comment by sougou** [comment]
I guess DCO check is just confused.

**Pull request merged**